src/asm/section.c

```diff
@@ -325,6 +335,7 @@ static struct Section *getSection(char const *name, enum SectionType type, uint3
 
 	sect->type = type;
 	sect->modifier = mod;
+	setSectionFilename(sect);
```

Comment:
Since `setSectionFilename` is only called once, and is only two lines, I would just set `sect->src` and `sect->fileLine` here with the other `sect` properties.